src/Drivers/DatabaseDriver.php

```diff
@@ -245,9 +278,13 @@ protected function resolveValue($feature, $scope)
      */
     public function set($feature, $scope, $value): void
     {
-        if (! $this->update($feature, $scope, $value)) {
-            $this->insert($feature, $scope, $value);
-        }
+        $this->newQuery()->upsert([
+            'name' => $feature,
+            'scope' => Feature::serializeScope($scope),
+            'value' => json_encode($value, flags: JSON_THROW_ON_ERROR),
+            static::CREATED_AT => $now = Carbon::now(),
+            static::UPDATED_AT => $now,
+        ], uniqueBy: ['name', 'scope'], update: ['value', 'updated_at']);
```

Comment:
When telling Pennant to set a feature, we can utilise `upsert` instead of attempting to update and then insert if there is nothing to update.

This also has a good impact on the number of database queries the library runs.